tests/Unit/UseCases/RestoreMembershipApplication/RestoreMembershipApplicationUseCaseTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace WMDE\Fundraising\MembershipContext\Tests\Unit\UseCases\RestoreMembershipApplication;
+
+use PHPUnit\Framework\TestCase;
+use WMDE\Fundraising\MembershipContext\Domain\Model\MembershipApplication;
+use WMDE\Fundraising\MembershipContext\Tests\Data\ValidMembershipApplication;
+use WMDE\Fundraising\MembershipContext\Tests\Fixtures\FakeApplicationRepository;
+use WMDE\Fundraising\MembershipContext\Tests\Fixtures\MembershipApplicationEventLoggerSpy;
+use WMDE\Fundraising\MembershipContext\UseCases\RestoreMembershipApplication\RestoreMembershipApplicationUseCase;
+
+/**
+ * @covers \WMDE\Fundraising\MembershipContext\UseCases\RestoreMembershipApplication\RestoreMembershipApplicationUseCase
+ * @covers \WMDE\Fundraising\MembershipContext\UseCases\RestoreMembershipApplication\RestoreMembershipApplicationResponse
+ */
+class RestoreMembershipApplicationUseCaseTest extends TestCase {
+
+	private const AUTH_USER_NAME = "Pintman Paddy Losty";
+
+	private MembershipApplicationEventLoggerSpy $membershipApplicationEventLogger;
+	private FakeApplicationRepository $applicationRepository;
+
+	protected function setUp(): void {
+		parent::setUp();
+		$this->membershipApplicationEventLogger = new MembershipApplicationEventLoggerSpy();
+		$this->applicationRepository = new FakeApplicationRepository();
+	}
+
+	public function testOnRestoreNonExistentMembershipApplication_actionFails() {
+		$useCase = $this->newUseCase();
+
+		$response = $useCase->restoreApplication( 1, self::AUTH_USER_NAME );
+
+		$this->assertFalse( $response->moderationChangeSucceeded() );
+	}
+
+	public function testOnRestoreMembershipApplication_actionSucceeds() {
+		$application = ValidMembershipApplication::newDomainEntity();
+		$application->cancel();
+		$useCase = $this->newUseCase( $application );
+
+		$response = $useCase->restoreApplication( 1, self::AUTH_USER_NAME );
```

Comment:
I don't like those "magic numbers" in tests, they are making them harder ro read/understand.

How about using `$application->getId()` in all of these tests, to make them more robust just in case the auto-assigned ID won't be 1 or rather give the ID more meaning? You could also use a constant with a comment like `Store will auto-assign ID 1 for new membership applications` to get around the static type check nagging because of the nullable ID.